gcore/rasterio.cpp

```diff
@@ -317,6 +318,17 @@ CPLErr GDALRasterBand::IRasterIO(GDALRWFlag eRWFlag, int nXOff, int nYOff,
             {
                 // Type to type conversion.
 
+#ifndef HAVE__FLOAT16
+                if (eBufType == GDT_Float16 || eBufType == GDT_CFloat16 ||
```

Comment:
If drivers are going to report Float16 for datasets even when the compiler doesn't support _Float16 we need to provide a way to users to get their data as Float32 for example, otherwise this is going to be a functional regression
(applies to a different number of places)

Reply:
I'm now using GFloat16 in all places in `gcore`.